lib/improver/nowcasting/accumulation.py

```diff
@@ -0,0 +1,214 @@
+        Returns:
+            cube_subsets (list of lists):
+                A list containing lists which comprise the rates cubes required
+                to construct a sequence of accumulation cubes that match the
+                period given by self.accumulation_period. If
+                self.accumulation_period is None then the whole list of cubes
+                is returned to give one accumulation across the whole set.
+        Raises:
+            ValueError: Accumulation period cannot be created from time
+                        intervals of input cubes.
+        """
+        # If no accumulation period is provided, calculate accumulation across
+        # all provided cubes.
+        if self.accumulation_period is None:
+            return [cubes]
+
+        fraction, cube_subset = np.modf(self.accumulation_period /
+                                        time_interval)
+
+        if fraction != 0:
+            msg = ("The specified accumulation period ({}) is not divisible "
+                   "by the time intervals between rates cubes ({}). As "
+                   "a result it is not possible to calculate the desired "
+                   "total accumulation period.".format(
+                       self.accumulation_period, time_interval))
+            raise ValueError(msg)
+
+        cube_subset = int(cube_subset)
+
+        # We expect 1 more rates cubes than accumulations cubes, if we have
+        # more or less, a targetted accumulation_period will be incomplete.
+        unused_cubes = int(len(cubes) % cube_subset)
+        if unused_cubes != 1 and cube_subset != 1:
+            msg = ("The provided cubes result in a partial period given the "
+                   "specified accumulation_period, i.e. the number of cubes "
+                   "is insufficient to give a set of complete periods. Only "
+                   "complete periods will be returned.")
+            warnings.warn(msg)
+
+        cube_subsets = []
+        for index in range(0, len(cubes) - unused_cubes, cube_subset):
+            cube_subsets.append(cubes[index:index + cube_subset + 1])
+        return cube_subsets
+
+    def process(self, cubes):
+        """
+        Calculate period precipitation accumulations based upon precipitation
+        rate fields.
+
+        Args:
+            cubes (iris.cube.CubeList):
+                A cubelist containing input precipitation rate cubes.
+        Returns:
+            accumulation_cubes (iris.cube.CubeList):
+                A cubelist containing precipitation accumulation cubes where
+                the accumulation periods are determined by plugin argument
+                accumulation_period.
+        """
+        # Standardise inputs to expected units.
+        cubes = enforce_coordinate_units_and_dtypes(
+            cubes, ['time', 'forecast_reference_time', 'forecast_period'],
+            inplace=False)
+        enforce_diagnostic_units_and_dtypes(cubes)
```

Comment:
OK great.  Can we note in the docstring which units we're converting to?  (I know this will be a standard; it also matters in this particular case, where it doesn't matter in other cases like eg neighbourhooding.)

Reply:
Done.